[samriddh2145] iP

samriddh2145 opened the pull request:
# Bing - Your Personal Task Manager

> “Efficiency is doing things right, Effectiveness is doing the right things.” – Peter Drucker

Bing helps you manage your tasks effectively, so you can focus on what truly matters. It's:

- **intuitive**
- **flexible**
- **FAST** and _efficient_ to use

All you need to do is:

1. Download it from [here](https://github.com/samriddh2145/ip).
2. Open the application.
3. Start adding your tasks.
4. Let Bing manage your tasks for you 😊.

---

### Features:
- [x] Manage to-do tasks
- [x] Set deadlines for tasks
- [x] Manage events with start and end times
- [x] Delete tasks
- [x] Find tasks 

### Code Example:
```java
public class Bing {

    private List<Task> tasks;

    public Bing() {
        this.tasks = new ArrayList<>();
    }

    public void addTask(Task task) {
        tasks.add(task);
    }

    public List<Task> getTasks() {
        return tasks;
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

samriddh2145 committed:
Level 0 - Rename main class and message.

samriddh2145 committed:
Level 0

samriddh2145 committed:
Level 1

samriddh2145 committed:
Level 2

samriddh2145 committed:
Level 3

samriddh2145 committed:
level4

samriddh2145 committed:
Automated_Testing

brianchoon commented on src/main/java/Bing.java:
```diff
@@ -0,0 +1,92 @@
+                System.out.println("______________________________\n");
+            } else if (input.startsWith("mark")) {
+                int x = Integer.parseInt(input.substring(5));
+                if (x>0 && x<=tasks.size()) {
```
Perhaps you may use some spacing around the characters here?

brianchoon commented on src/main/java/Bing.java:
```diff
@@ -0,0 +1,92 @@
+
+            } else if (input.startsWith("unmark")) {
+                int x = Integer.parseInt(input.substring(7));
+                if (x>0 && x<=tasks.size()) {
```
Perhaps you may use some spacing around the characters here too?

brianchoon commented on src/main/java/Bing.java:
```diff
@@ -0,0 +1,92 @@
+        String message = "______________________________\n"
+                + "Hi! My name is Bing\n"
+                + "How can I help you?\n"
+                + "______________________________\n";
```
I notice you have many areas in your code where you are using a string of underscores to denote a line. You may consider declaring a variable, e.g. LINE, to avoid code repetition!

brianchoon commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,15 @@
+        this.to = to;
+    }
+
+
```
Nit: Unnecessary newline

brianchoon reviewed (commented):
Overall, I found your code easy to read. Perhaps you may have repeated some code, which you can easily fix!

samriddh2145 committed:
Level5

samriddh2145 committed:
Level6

samriddh2145 committed:
Enums

starbucksventi commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,21 @@
+    private String info;
+    private TaskStatus isDone;
+
+    public Task(String info) {
```
Consider adding a JavaDoc comment to the explain constructor for this class.

starbucksventi commented on src/main/java/TaskStatus.java:
```diff
@@ -0,0 +1,16 @@
+        this.StatusSymbol = StatusSymbol;
+    }
+
+    public String getStatusSymbol() {
```
Consider adding a Javadoc comment for this. Specifically, what is the status that is returned about.

starbucksventi commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+
+    private String deadline;
+
+    public Deadline(String info, String deadline) {
```
Consider adding a JavaDoc comment for this as it will likely cause a Checkstyle error, since public constructors are required to have such a comment.

starbucksventi commented on src/main/java/Bing.java:
```diff
@@ -0,0 +1,144 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+public class Bing {
+    public static void main(String[] args) {
```
Consider abstracting away portions of your code to helper methods and then refactoring your code to match the OOP style of coding. Kudos on the extensive error handling nonetheless!

starbucksventi reviewed (commented):
Overall, some refactoring & adding of JavaDoc comments seems to be required for code readability. Otherwise, looks good to merge!

samriddh2145 committed:
Implemented saving and loading file functionality

samriddh2145 committed:
Merge branch 'branch-level-7'

samriddh2145 committed:
Chatbot now saves date and time as LocalDateTime

samriddh2145 committed:
Merge branch 'branch-Level-8'

samriddh2145 committed:
Added more OOP properties

samriddh2145 committed:
Added package

samriddh2145 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

samriddh2145 committed:
Add JUnit Tests

samriddh2145 committed:
Add JavaDoc comments

samriddh2145 committed:
Apply coding standards

samriddh2145 committed:
Implement Level-9

samriddh2145 committed:
Resolve conflicts and merge branch-A-CodingStandard

samriddh2145 committed:
Resolve conflicts and merge branch-Level-9

samriddh2145 committed:
Add GUI and implement Level-10

samriddh2145 committed:
Merge branch 'branch-Level-10'

samriddh2145 committed:
Add assert feature

samriddh2145 committed:
Improve code quality

samriddh2145 committed:
Merge pull request #2 from samriddh2145/branch-A-Assertions

Add features to A-Assertions

samriddh2145 committed:
Merge pull request #3 from samriddh2145/branch-A-CodeQuality

Add changes to improve code quality

samriddh2145 committed:
Add stats command to display task statistics

Implemented a new 'stats' command that provides users with insights into their tasks. The statistics include the total number of tasks, the number of marked and the number of unmarked tasks.

samriddh2145 committed:
Fix the issue of welcome message

samriddh2145 committed:
Add User Guide and Product Image

samriddh2145 committed:
Update README

samriddh2145 committed:
Add changes for jar release

samriddh2145 committed:
Add Launcher class
